### PR TITLE
455 configuration edit UI

### LINKED ISSUE #455: Add UI for editing category config
i.e. the different drug forms, routes etc.  

These should be a predefined dropdown list in the drug edit form, but from a different screen we want to be able to add/edit/remove options! 

Also to be considered - when these options are selected, should we store a reference to them on the drug (e.g. so if a typo is fixed in this new page, that gets dispersed to all the drugs? Or just copy the string, so that we can edit these options freely without worrying to support old ones.

### PR DESCRIPTION
**Pull request by lache-melvin**
Fixes #455

## Description

- Adds UI for managing config
- Tabs for managing categories (i.e. Routes/Forms)
- Tab for managing properties
  - Have not implemented delete for properties, as these are used as reference data for other code... but could implement a deactivate or something in future if it is desired
  - Is the helper text in the properties add/edit modal ok?

Screenshot 2023-12-20 at 12 34 45 PM
Screenshot 2023-12-20 at 12 34 54 PM
Screenshot 2023-12-20 at 12 35 08 PM

### CODE REVIEW

**Comment by jmbrunskill** [comment]
> * Is the helper text in the properties add/edit modal ok?

Yup looks good. I'm slightly worried about using {{code}} for non universal codes. Seems slightly confusing to me, but let's check with others...

File: frontend/system/src/Admin/Configuration/CategoryConfigTab.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  label: string;
+  value: string;
+};
+
+type CategoryConfigTabProps = {
+  data: CategoryOption[];
+  category: string;
+};
+
+const CategoryConfigTabComponent = ({
```
Comment (jmbrunskill) [inline frontend/system/src/Admin/Configuration/CategoryConfigTab.tsx:35]: The name `CategoryConfigTab` confused me a little bit, as we have categories for `Drug` and `Consumable`
Maybe `ListConfigTab`? Can't think of a perfect name :)

Comment (jmbrunskill) [reply frontend/system/src/Admin/Configuration/CategoryConfigTab.tsx:35]: The `Categories` are `types` in the dgraph but that's also confusing.
If we're considering changing a lot about the dgraph, we could make category the thing like `drug`, `unit` etc?
Maybe that's another approach...

**Review by jmbrunskill — APPROVED** [review]
Looks good, will be good to try it out properly once saving is implemented.

**Comment by jmbrunskill** [comment]
It's not a major as it won't be used a lot, but the delete feels very disconnected from the selection box...
image

**Comment by lache-melvin** [comment]
> Yup looks good. I'm slightly worried about using {{code}} for non universal codes. Seems slightly confusing to me, but let's check with others...

Ah yeah good point, we can ask around tomorrow?

File: frontend/system/src/Admin/Configuration/CategoryConfigTab.tsx (hunk 1 above)
Comment (lache-melvin) [reply frontend/system/src/Admin/Configuration/CategoryConfigTab.tsx]: Gone with `OptionList`... not sure I love that name either but at least I don't think its in conflict with anything else 😅  [a0c6cbe](https://github.com/msupply-foundation/unified-codes/pull/462/commits/a0c6cbe20854227bf55890d6b876cde11fd1e1c8)

**Comment by lache-melvin** [comment]
> It's not a major as it won't be used a lot, but the delete feels very disconnected from the selection box... image

Yeah... I tried putting it within that main grey box but it looked a bit strange, pushing the rest of the content down... keen for any ideas - would also be great to get Anisha's eye over the app in general at some point in the new year!

**Pull request merged**